### PR TITLE
Import copy fastpath

### PR DESCRIPTION
**Pull request by cgwalters**
Splitting the copy fastpath bit out from https://github.com/ostreedev/ostree/pull/1193

### CODE REVIEW

File: src/libostree/ostree-repo-commit.c [hunk 1]
```diff
@@ -3230,76 +3226,142 @@ copy_detached_metadata (OstreeRepo    *self,
...
           *out_was_supported = FALSE;
           return TRUE;
         }
-      else
-        g_assert_not_reached ();
-    }
 
-  if (!_ostree_repo_ensure_loose_objdir_at (self->objects_dir_fd, loose_path_buf, cancellable, error))
-    return FALSE;
+      /* This is yet another variation of glnx_file_copy_at()
```
Comment (jlebon) [inline src/libostree/ostree-repo-commit.c]: I'm confused, if it's a bare file, we *have* to `chown`, right? (Well, unless we're running as the id of the source object, which is probably an optimization we should teach to `glnx_file_copy_at`).

Comment (cgwalters) [reply src/libostree/ostree-repo-commit.c]: Yep, realized that after submitting. Will fix soon.

Comment (cgwalters) [reply src/libostree/ostree-repo-commit.c]: (To clarify we only need to chown for bare repos)

**Comment by cgwalters** [comment]
bot, retest this please

**Comment by jlebon** [comment]
@rh-atomic-bot r+ e9d7273

**Pull request closed without merging**